tests/unit/dataset/loader/test_exgentic.py

```
@@ -0,0 +1,422 @@
+    source: str, expected: str
+) -> None:
+    assert canonical_source_model(source) == expected
+
+
+def test_invalid_filter_lists_typed_values() -> None:
+    with pytest.raises(DatasetLoaderError, match=r"available filters:.*Kimi-K2.5"):
+        _loader({"source_model": "unknown"})
+
+
+def test_unsupported_filter_pair_fails_before_loading() -> None:
+    with pytest.raises(DatasetLoaderError, match="Unsupported.*available source"):
```

Ruff (0.15.18)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)